### ISSUE TITLE
validation error: `Optional[Mapping[str, str]]`

### ISSUE THREAD

**Issue by slippycheeze** [seandstewart/typical#83]
* typical version: `2.0.12`
* Python version: 
* Operating System: macOS 15

### Description
Looks like optional mapping validation in non-strict mode fails to compile.  The data being checked is irrelevant, problem happens during type compilation.

```python
from typing import *
import typic
typic.validate(Optional[Mapping[str, str]], {})
```
Note: I'd expect the minimal repro to validate, but this is about the type validation compiler, not the `validate` call itself -- that is just the easy way to trigger it.  I ran into it in an annotated function where one argument was `Optional[Mapping[Strict[str], str]]`, but cut it down to the above.

`Optional[Mapping]` will succeed, so it seems to be an assumption in the key/value validation code for `Mapping` that didn't expect the `Union` type above it or something.

```
] python3 -c 'from typing import *; import typic; typic.validate(Optional[Mapping[str, str]], None)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/slippycheeze/homebrew/lib/python3.7/site-packages/typic/serde/resolver.py", line 120, in validate
    resolved: SerdeProtocol = self.resolve(annotation)
  File "/Users/slippycheeze/homebrew/lib/python3.7/site-packages/typic/serde/resolver.py", line 458, in resolve
    deserializer, validator = self.des.factory(anno, constraints)
  File "/Users/slippycheeze/homebrew/lib/python3.7/site-packages/typic/serde/des.py", line 527, in factory
    deserializer = self._build_des(annotation)
  File "/Users/slippycheeze/homebrew/lib/python3.7/site-packages/typic/serde/des.py", line 438, in _build_des
    self._build_mapping_des(func, anno_name, annotation)
  File "/Users/slippycheeze/homebrew/lib/python3.7/site-packages/typic/serde/des.py", line 324, in _build_mapping_des
    if issubclass(annotation.origin, defaultdict):
TypeError: issubclass() arg 1 must be a class
```

**Comment by seandstewart** [comment]
Hmm, this looks to be a regression. I’ll take a look in the morning.

**Comment by seandstewart** [comment]
This is fixed in [v2.0.14](https://github.com/seandstewart/typical/releases/tag/v2.0.14)

**Comment by slippycheeze** [comment]
Thanks so much for being so responsive!
